File: montecarlo.nim
import math

type Coord = tuple[ x:float, y:float ]

proc getrand():float = random(1.0)

proc in_qsect(coord:Coord):bool =
  result = pow(coord.x, 2) + pow(coord.y, 2) <= 1


const ITERATION:int = 800_000

randomize()

var count_in_qsect:int = 0

for i in 1..ITERATION:
  if in_qsect((x: getrand(), y:getrand())):
    inc(count_in_qsect)

echo("In quarter sector = ", count_in_qsect, " / ", ITERATION)
echo("Estimated Pi: ", count_in_qsect * 4 / ITERATION)

File: montecarlo.lua
#!/usr/bin/env lua

function getrand()
  return math.random()
end

function in_qsect(coord)
  return math.pow(coord.x, 2) + math.pow(coord.y, 2) <= 1
end

iteration = 800000

count_in_qsect = 0
for i = 1, iteration do
  if in_qsect({ x = getrand(), y = getrand() }) then
    count_in_qsect = count_in_qsect + 1
  end
end

print(("In quarter sector = %d / %d"):format(count_in_qsect, iteration))
print("Estimated Pi: " ..  count_in_qsect * 4 / iteration)

